### PR TITLE
treat suction ranges independently

### PR DESCRIPTION
**Pull request by mlautman**

### CODE REVIEW

File: src/grasp_generator.cpp [hunk 1]
```diff
@@ -1046,50 +1050,107 @@ bool GraspGenerator::generateSuctionGrasps(const Eigen::Affine3d& cuboid_top_pos
...
   // We define min, max and inc for each for loop here for readability
-  double x_min = -depth / 2.0 + grasp_data->active_suction_range_x_ / 2.0;
-  double x_max = depth / 2.0 - grasp_data->active_suction_range_x_ / 2.0;
-  double x_inc = grasp_data->grasp_resolution_;
 
-  double y_min = -width / 2.0 + grasp_data->active_suction_range_y_ / 2.0;
-  double y_max = width / 2.0 - grasp_data->active_suction_range_y_ / 2.0;
-  double y_inc = grasp_data->grasp_resolution_;
+  // if X range is less than y range then we use x range for the xy range
+  double xy_min, xy_max, xy_inc;
```
Comment (davetcoleman) [inline src/grasp_generator.cpp]:
```suggestion
  double xy_min, xy_max, xy_increment;
```

let's not over abbreviate, that makes for very difficult code to read

File: src/grasp_generator.cpp [hunk 2]
```diff
@@ -1046,50 +1050,107 @@ bool GraspGenerator::generateSuctionGrasps(const Eigen::Affine3d& cuboid_top_pos
...
+  {
+    xy_min = -depth / 2.0 + grasp_data->active_suction_range_x_ / 2.0;
+    xy_max = depth / 2.0 - grasp_data->active_suction_range_x_ / 2.0;
+    xy_inc = grasp_data->grasp_resolution_;
+  }
+  else
+  {
+    xy_min = -width / 2.0 + grasp_data->active_suction_range_y_ / 2.0;
+    xy_max = width / 2.0 - grasp_data->active_suction_range_y_ / 2.0;
+    xy_inc = grasp_data->grasp_resolution_;
```
Comment (davetcoleman) [inline src/grasp_generator.cpp]: does line 1064 and 1070 need to be inside the if statement?

File: src/grasp_generator.cpp [hunk 3]
```diff
@@ -1046,50 +1050,107 @@ bool GraspGenerator::generateSuctionGrasps(const Eigen::Affine3d& cuboid_top_pos
...
+      grasp_pose = grasp_poses[i].rotation() * grasp_pose;
+      grasp_pose.translation() += grasp_poses[i].translation();
 
-          addGrasp(grasp_pose, grasp_data, grasp_candidates, cuboid_top_pose, 0);
-        }
-      }
+      grasp_poses.push_back(grasp_pose);
+    }
+  }
+
```
Comment (davetcoleman) [inline src/grasp_generator.cpp]: add comments for these sections e.g. "Generate Y suction grasps" ?

Comment (mlautman) [reply src/grasp_generator.cpp]: done

File: src/grasp_generator.cpp [hunk 4]
```diff
@@ -798,21 +802,23 @@ double GraspGenerator::scoreGrasp(const Eigen::Affine3d& grasp_pose, const Grasp
...
+                             << ",\t" << ideal_grasp_pose_.translation().z() << ")\t("
+                             << ideal_grasp_pose_.rotation().eulerAngles(0, 1, 2)(0) << ",\t"
+                             << ideal_grasp_pose_.rotation().eulerAngles(0, 1, 2)(1) << ",\t"
                              << ideal_grasp_pose_.rotation().eulerAngles(0, 1, 2)(2) << ")");
 
   // get portion of score based on the orientation
   Eigen::Vector3d orientation_scores = GraspScorer::scoreRotationsFromDesired(grasp_pose, ideal_grasp_pose_);
 
   // get portion of score based on the translation
+  // Eigen::Vector3d translation_scores =
```
Comment (henningkayser) [inline src/grasp_generator.cpp]: Will this be added later?

File: src/grasp_generator.cpp [hunk 5]
```diff
@@ -978,16 +983,17 @@ bool GraspGenerator::generateSuctionGrasps(const Eigen::Affine3d& cuboid_top_pos
...
   setIdealGraspPose(ideal_grasp);
 
-  bool debug_first_transform = false;
-
-  if (debug_top_grasps_ && debug_first_transform)
+  if (debug_top_grasps_)
   {
-    visual_tools_->prompt("publish cuboid pose");
-    visual_tools_->publishAxis(cuboid_center_top_grasp, rviz_visual_tools::MEDIUM, "cuboid_center_top_grasp");
+    // visual_tools_->prompt("publish cuboid pose");
```
Comment (henningkayser) [inline src/grasp_generator.cpp]: I would remove all commented prompts if they are not needed.

File: src/grasp_generator.cpp [hunk 6]
```diff
@@ -1032,61 +1040,126 @@ bool GraspGenerator::generateSuctionGrasps(const Eigen::Affine3d& cuboid_top_pos
...
+
+  num_grasps = grasp_poses.size();
+  for (std::size_t i = 0; i < num_grasps; ++i)
+  {
+    addGrasp(grasp_poses[i], grasp_data, grasp_candidates, cuboid_top_pose, 0);
+    if (debug_top_grasps_)
+    {
+      visual_tools_->publishAxis(grasp_poses[i], rviz_visual_tools::MEDIUM, "pose");
+      Eigen::Affine3d ideal_copy(ideal_grasp_pose_);
+      ideal_copy.translation() = Eigen::Vector3d(0.7, 0.3, 1.25);
```
Comment (henningkayser) [inline src/grasp_generator.cpp]: Where do these values come from? Looks like they should be parameters

File: src/grasp_generator.cpp (hunk 4 above)
Comment (mlautman) [reply src/grasp_generator.cpp]: No, this is the finger grasp scoring method. I deleted the dead code

File: src/grasp_generator.cpp (hunk 5 above)
Comment (mlautman) [reply src/grasp_generator.cpp]: done

File: src/grasp_generator.cpp (hunk 6 above)
Comment (mlautman) [reply src/grasp_generator.cpp]: Just some random translation to make it easy to see. I have made it more clear and less magical

**Comment by mlautman** [comment]
@henningkayser @davetcoleman ping pong

**Pull request merged**